include/libdnf/common/set.hpp

```diff
@@ -39,13 +87,15 @@ class Set {
 
     // GENERIC OPERATIONS
     bool empty() const noexcept { return data.empty(); };
-    std::size_t size() const noexcept { return data.size(); }
+    size_type size() const noexcept { return data.size(); }
     void clear() noexcept { data.clear(); }
 
     // ITEM OPERATIONS
-    void add(const T & obj) { data.insert(obj); }
-    void add(T && obj) { data.insert(std::move(obj)); }
-    void remove(const T & obj) { data.erase(obj); }
+    bool add(const T & obj) { return data.insert(obj).second; }
```

Comment:
As I see you improve Python standard behaviour. In python set().add("hello") returns None. Anyway I like it.